ui/src/app/pages/admin/admin-institution-edit.tsx

```diff
@@ -0,0 +1,411 @@
+    const {institution} = this.state;
+    return institution.displayName || institution.organizationTypeOtherText ||
+        institution.organizationTypeEnum || institution.duaTypeEnum || institution.emailAddresses || institution.emailDomains;
+  }
+
+  get fieldsNotEditedEditInstitution() {
+    const {institution} = this.state;
+    return institution.displayName === institutionToEdit.displayName &&
+        institution.organizationTypeEnum === institutionToEdit.organizationTypeEnum &&
+        institution.duaTypeEnum === institutionToEdit.duaTypeEnum &&
+        institution.emailAddresses === institutionToEdit.emailAddresses &&
+        institution.emailDomains === institutionToEdit.emailDomains &&
+        institution.userInstructions === institutionToEdit.userInstructions &&
+        institution.organizationTypeOtherText === institutionToEdit.organizationTypeOtherText;
+  }
+
+
+  noEmptyRequiredFields() {
+    const {institution} = this.state;
+    let emailValid = true;
+    if (institution.duaTypeEnum) {
+      emailValid = institution.duaTypeEnum === DuaType.MASTER ?
+          institution.emailDomains !== undefined : institution.emailAddresses !== undefined;
+    }
+    return !emailValid || !institution.displayName || !institution.organizationTypeEnum ||
+      !institution.duaTypeEnum ||
+        (institution.organizationTypeEnum === OrganizationType.OTHER &&
+            !institution.organizationTypeOtherText);
+  }
+
+  // Disable save button if
+  // a) No fields were edited or if there are any errors
+  // b) email address/Domain are not valid
+  // c) Required fields are not empty
+  disableSave(errors) {
+    return this.noEmptyRequiredFields() || (errors && errors.displayName) || this.fieldsNotEdited()
+      || this.state.invalidEmailAddress || this.state.invalidEmailDomain;
+  }
+
+  async saveInstitution() {
+    const {institution} = this.state;
+    if (institution) {
+      this.setState({invalidEmailAddress: false});
+      if (institution.duaTypeEnum === DuaType.MASTER) {
+        institution.emailAddresses = [];
+      } else {
+        institution.emailDomains = [];
+      }
+      if (institution.organizationTypeEnum !== OrganizationType.OTHER) {
+        institution.organizationTypeOtherText = null;
+      }
+    }
+    if (this.props.urlParams.institutionId) {
+      await institutionApi().updateInstitution(this.props.urlParams.institutionId, institution)
+        .then(value => this.backNavigate())
+        .catch(reason => this.handleError(reason));
+    } else {
+      await institutionApi().createInstitution(institution)
+        .then(value => this.backNavigate())
+        .catch(reason => this.handleError(reason));
+    }
+  }
+
+  handleError(rejectReason) {
+    let errorMsg = 'Error while saving Institution. Please try again later';
+    if (rejectReason.status === 409) {
+      errorMsg  = 'Institution with Name ' + this.state.institution.displayName + ' already exist';
```

Comment:
We can't say for certain that this is the reason.  Better to use the error text which is returned here.